File: app/src/main/java/com/example/monappliimmo/BatimentAdapter.java
package com.example.monappliimmo;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class BatimentAdapter extends RecyclerView.Adapter<BatimentAdapter.BatimentViewHolder> {

    private List<Batiment> listeBatiments;

    public BatimentAdapter(List<Batiment> listeBatiments) {
        this.listeBatiments = listeBatiments;
    }

    @NonNull
    @Override
    public BatimentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_batiment, parent, false);
        return new BatimentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BatimentViewHolder holder, int position) {
        Batiment batiment = listeBatiments.get(position);
        holder.tvAdresse.setText(batiment.getAdresse());
        holder.tvVille.setText(batiment.getVille());

        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(v.getContext(), BatimentDetailActivity.class);
            intent.putExtra("BATIMENT_OBJ", batiment);
            v.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return listeBatiments.size();
    }

    public static class BatimentViewHolder extends RecyclerView.ViewHolder {
        TextView tvAdresse, tvVille;

        public BatimentViewHolder(@NonNull View itemView) {
            super(itemView);
            tvAdresse = itemView.findViewById(R.id.tvAdresseBatiment);
            tvVille = itemView.findViewById(R.id.tvVilleBatiment);
        }
    }
}

File: app/src/main/java/com/example/monappliimmo/BatimentAdapter.kt
package com.example.monappliimmo

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class BatimentAdapter(
    private val listeBatiments: List<Batiment>
) : RecyclerView.Adapter<BatimentAdapter.BatimentViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BatimentViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_batiment, parent, false)
        return BatimentViewHolder(view)
    }

    override fun onBindViewHolder(holder: BatimentViewHolder, position: Int) {
        val batiment = listeBatiments[position]

        holder.tvAdresse.text = batiment.adresse
        holder.tvVille.text = batiment.ville

        holder.itemView.setOnClickListener { v ->
            val context = v.context
            val intent = Intent(context, BatimentDetailActivity::class.java)

            intent.putExtra("BATIMENT_OBJ", batiment)

            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return listeBatiments.size
    }

    class BatimentViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvAdresse: TextView = itemView.findViewById(R.id.tvAdresseBatiment)
        val tvVille: TextView = itemView.findViewById(R.id.tvVilleBatiment)
    }
}